datahub-actions/src/datahub_actions/plugin/source/acryl/datahub_cloud_event_source.py

```diff
@@ -0,0 +1,202 @@
+import json
+import logging
+import time
+from dataclasses import dataclass
+from typing import Iterable, List, Optional
+
+from datahub.configuration import ConfigModel
+from datahub.emitter.serialization_helper import post_json_transform
+
+# DataHub imports.
+from datahub.metadata.schema_classes import GenericPayloadClass
+
+from datahub_actions.event.event_envelope import EventEnvelope
+from datahub_actions.event.event_registry import (
+    ENTITY_CHANGE_EVENT_V1_TYPE,
+    EntityChangeEvent,
+)
+
+# May or may not need these.
+from datahub_actions.pipeline.pipeline_context import PipelineContext
+from datahub_actions.plugin.source.acryl.datahub_cloud_events_ack_manager import (
+    AckManager,
+)
+from datahub_actions.plugin.source.acryl.datahub_cloud_events_consumer import (
+    DataHubEventsConsumer,
+    ExternalEvent,
+)
+from datahub_actions.source.event_source import EventSource
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+ENTITY_CHANGE_EVENT_NAME = "entityChangeEvent"
+
+
+# Converts a DataHub Events Message to an EntityChangeEvent.
+def build_entity_change_event(payload: GenericPayloadClass) -> EntityChangeEvent:
+    return EntityChangeEvent.from_json(payload.get("value"))
```

Comment:
does this need any try/catch?

Reply:
no - this should raise an exception if it fails to parse. all i would do here is wrap and raise the exception. i can do that to add context but this indicates a fundamental issue